### ISSUE TITLE
seems like running into dead loop

### ISSUE THREAD

**Issue by fffairforce** [fffairforce/BME547-TSH-Test-Data-Conversion#2]
https://github.com/fffairforce/BME547-TSH-Test-Data-Conversion/blob/f3695c44c4c90da883a757061aef2663d9a84c35/dictionary.py#L102-L112

Dr ward,

would you see if you can have some advice. This takes forever to run and theres no warning during debug.... I don't know how to fix it.

**Comment by dward2** [comment]
Here is where a debugger is useful.  I ran your your code line by line in a debugger.  The endless loop is not in the diagnose procedure.  It is actually in this while loop:
https://github.com/fffairforce/BME547-TSH-Test-Data-Conversion/blob/f3695c44c4c90da883a757061aef2663d9a84c35/dictionary.py#L57-L69
The question is, why does this loop not end?  Run it through a debugger.  Think about it and let me know if you cannot figure it out.

**Comment by fffairforce** [comment]
How can I debug by line not defining the input(person_info) here. I simply drag the module out and cant be debuged. Or I should make up a simple list for it?

**Comment by dward2** [comment]
You shouldn't have to change anything to debug the code.  If you have PyCharm or another IDE, set a breakpoint a line 57.  Then, start the program in Debug mode.  It will pause at the start of the loop.  You can then start the debugger to go line-by-line  and see what is happening.  If you don't have an IDE, you can use `pudb` as we discussed in class on Wednesday.

**Comment by dward2** [comment]
Another suggestion:  compare the while loop above with the while loop found at:
https://github.com/fffairforce/BME547-TSH-Test-Data-Conversion/blob/f3695c44c4c90da883a757061aef2663d9a84c35/dictionary.py#L86-L99

Both loop through a list, but this one (lines 86 through 99) works while the other one  from above (lines 57-69) doesn't.

**Comment by fffairforce** [comment]
gotcha, thanks!

**Comment by fffairforce** [comment]
ohh there's another problem I didn't put the txt file down the folder of IDE or interpretor(spyder) so I can't actually open it directly but the final code shouldn't be containing my local location... It kind of back and forth, Is there a way to avoid it?

**Comment by dward2** [comment]
I believe you are asking:  how do I avoid having my specific file system path in my code, such as
```
in_file = open("C:\User\daw74\files\BME547\repos\assignment4\test_data.txt", "r")
```
Is that correct?  If so, the easiest way is to have the test_data.txt file be in the same folder as your python code.  Then, you can just reference the file as "test_data.txt".  Or, if you want to have a subfolder that contains your data, you could just reference the file as "subfolder_name/test_data.txt".

Does that answer your question?  Or, were you asking something else?

**Comment by fffairforce** [comment]
Yes, that's what Im lookoing for, Cuz I can't find where I install my spyder.... the folder is gone. I will do it in pycharm

**Comment by fffairforce** [comment]
Prof,
why is this pycharm so user unfriendly pushing files? I cannot only push a single adjusted file? It's like even I clickde off the rest commits there is still hundreds of new file said added to commit....Now I am not share to put this on git hub. It'll totally be a mess
And also, I test pep8 with it and no warnings but once a pushed up the travis-ci just gave me a red cross.

**Comment by dward2** [comment]
Remember, PyCharm is just a tool.  If you are having trouble with it, use Git Bash and the command line for doing your git work, and just use PyCharm as a text editor.  You do not have to use PyCharm to do your git work.  Same for PEP8.  Verify that your code meets PEP-8 standards by running `pytest` from the command line.

**Comment by fffairforce** [comment]
ok....that stuff is driving me mad... I'm going to leavto and use basic ways

**Comment by dward2** [comment]
Also, I just looked at one of your recent Travis builds and it looks like the two PEP8 errors you are mostly getting are the line length being too long and having tabs instead of spaces.  There are some settings in PyCharm that might help with those.  

As described at https://github.com/dward2/BME547/blob/master/Resources/PyCharm/PycharmSettings.md#editor--code-style, you can set the line length warning in PyCharm by opening the Settings window, then in the left-hand panel, click on Editor/Code Style.  An option will come up in the right-hand panel saying "Hard Wrap at".  Set that at 79.

Then, under Code Style there is another submenu called Python.  On the Tabs and Indents tab, make sure that Use tab character is unchecked.

**Comment by fffairforce** [comment]
all right I will check on it!

**Comment by fffairforce** [comment]
So, since the previous operations I did now my branch isn't clear enough and I cannot actually do travis-ci online test for single file once a time(if you would check the commit) Will it be an issue to lose point, when I upload the last file I can get all of the files passed by then.

**Comment by dward2** [comment]
I'm not 100% sure what you are asking.  If you can get your final commit on passing on Travis-CI, that is the primary issue.  For now, all that Travis is running is the PEP-8 check on your two files.  It is not actually running the tests inside your file `TSH-test.py`.  That is because `TSH-test.py` does not have the correct name to be automatically found by pytest.  The filename needs to start with `tsh_`.  So, I would rename your testing file `test_dictionary.py`.  Then, pytest will discover this file and run the tests within it.

**Comment by fffairforce** [comment]
I see.

**Comment by fffairforce** [comment]
I got a syntax dummy question but still a bit puzzled. Cuz I'm now running the whole stuff which I should have done from the first place......
Do I actually have to quote every time in each module if I am using the parameters in another module.
e.g.
![image](https://user-images.githubusercontent.com/42806161/53816778-30229c00-3f32-11e9-9c49-04fabd8a0b51.png)
![image](https://user-images.githubusercontent.com/42806161/53816885-63fdc180-3f32-11e9-8d46-25720d494ce5.png)
in this case I import the signal first and it can work as I replace `data` with `signal`
![image](https://user-images.githubusercontent.com/42806161/53816927-7bd54580-3f32-11e9-95d6-211980a4e6f1.png)
if not replacing there will be a error:
![image](https://user-images.githubusercontent.com/42806161/53817102-c8b91c00-3f32-11e9-9ef5-7ea3ef75e2fc.png)

**Comment by fffairforce** [comment]
I know I will have to do `time, signal = data_in()`
But not understanding why can't I quote a parameter in function `data_in` as I tended to do

**Comment by dward2** [comment]
Please re-post your question as an issue in your ECG assignment repository, with a permalink to the code in questions.  I need to see your entire code to answer your question.  Thanks.
